specs/jsonschema-core.md

```diff
@@ -950,40 +949,32 @@ an [absolute IRI](https://www.rfc-editor.org/rfc/rfc3987.html#section-2.2)
-particular structural location. This allows a subschema to be relocated without
-requiring JSON Pointer references to be updated.
-
-The `$anchor` and `$dynamicAnchor` keywords are used to specify such fragments.
-They are identifier keywords that can only be used to create plain name
-fragments, rather than absolute IRIs as seen with `$id`.
-
-`$anchor` defines a reference target for `$ref`. The fragment defined by this
-keyword is appended to the IRI of the schema resource containing it. As
-discussed in {{id-keyword}}, this is either the nearest `$id` in the same or an
-ancestor schema object, or the base IRI for the document as determined according
-to [RFC 3987][rfc3987] and
-[RFC 3986][rfc3986].
-
-In contrast, `$dynamicAnchor` operates independently of resource IRIs and is
-instead dependent on the dynamic scope of the evaluation. `$dynamicAnchor`
-defines a reference target for the `$dynamicRef` keyword. This advanced feature
-makes it easier to extend recursive schemas such as the meta-schemas, without
-imposing any particular semantics on that extension. See {{dynamic-ref}} for
-details.
-
-In most cases, the normal fragment behavior both suffices and is more intuitive.
-Therefore it is RECOMMENDED that `$anchor` be used to create plain name
-fragments unless there is a clear need for `$dynamicAnchor`.
+Using JSON Pointers in IRI fragments to reference subschemas couples the IRI to
+the structure of the schema. Using plain name fragment identifiers in IRI
+fragments to identify subschemas is sometimes preferable because it is not tied
+to a particular structural location. This allows a subschema to be relocated
+without requiring references to be updated.
 
-If present, the value of these keywords MUST be a string and MUST conform to the
-plain name fragment identifier syntax defined in {{fragments}}.[^4]
+The `$anchor` and `$dynamicAnchor` keywords are used to define
+location-independent identifiers for subschemas within a schema resource.
+
+`$anchor` defines a plain name fragment identifier that can be used in IRI
+fragments as an alternative to JSON Pointers.[^4] See {{fragments}}.
 
 [^4]: Note that the anchor string does not include the "#" character, as it is
-not a IRI reference. An `$anchor`: "foo" becomes the fragment `#foo` when used
-in a IRI. See below for full examples.
+just a fragment identifier not an IRI reference. To reference the "foo"
+`$anchor` from the same Schema Resource, you would use the fragment-only IRI
+`#foo`. See below for full examples.
+
+`$dynamicAnchor` defines a different kind of identifier that only has meaning
+when used as a value for `$dynamicRef`. It's not a fragment identifier that can
+be used in an IRI. `$dynamicRef` uses a syntax similar to a fragment-only IRI,
+but the semantics are different. Unlike IRIs, the fragment identifies the
+primary resource (the schema resource) in addition to the secondary resource
+(the subschema within that schema resource). See {{dynamic-ref}} for details.
```

Comment:
If you're looking to distinguish a `$dynamicAnchor` fragment identifier from an `$anchor` fragment identifier, I don't think this is clear enough.  An IRI that has a fragment identifier also "identifies the primary resource (the schema resource) in addition to the secondary resource (the subschema within that schema resource)".

If that's not your intent, then I'm not sure what purpose this sentence serves.

Reply:
The sentence isn't referring to an IRI that has a fragment, it's referring to just the fragment or the IRI or IRI-like value.

In `"$ref": "example#foo"`, "example" determines the primary resource and "foo" determines the secondary resource. In `"$dynamicRef": "#foo"`, "foo" determines both the primary resource and the secondary resource (which is why the part before the `#` is always empty).

I think sentence is clear that it's referring to the fragment of the IRI, not the whole IRI, but let me know if you think there's a better way to make that more clear.

Reply:
I think it's the "Unlike IRIs" part that throws it off.  Why make the comparison, especially when IRIs (ones with fragments, anyway) also do the thing you're saying these fragments do?

Reply:
I tried making a change. Let me know if you think it help. Honestly, I'm having trouble seeing what you're seeing in the sentence, so I'm not sure how to make it better.

> Why make the comparison, especially when IRIs (ones with fragments, anyway) also do the thing you're saying these fragments do?

This sentence is talking about the fragments in IRIs, not the whole of IRI capabilities. Is that the confusion?